### DISCUSSION TITLE
Declarative User Profile schema admin API for attributes?

### DISCUSSION THREAD

**Discussion by sgpinkus** [keycloak/keycloak#21527] [Q&A] [answered]
I've enabled Declarative User Profile and added some attribute schema. I want to automate this process via kcadm.sh / API. I noticed there is a JSON schema editor. Basically what I want to do is upload these schema via the API. I couldn't find any doc on such an API. Is it possible?

**Comment by ssilvert** [comment, marked as answer]
It's definitely possible to do it with the admin REST API as we use these admin endpoints in the admin console.  Not sue of all the details, but this might get you started:

https://github.com/keycloak/keycloak/blob/main/js/apps/admin-ui/src/realm-settings/user-profile/JsonEditorTab.tsx

**Comment by jbman** [comment]
You can check what the Admin Console is doing with browser dev tools:
PUT	/admin/realms/{realm}/users/profile

It's also listed at https://www.keycloak.org/docs-api/21.0.2/rest-api/#_users_resource

**Comment by sgpinkus** [comment]
FYI it was like:

```
kcadm.sh update users/profile -f declarative-user-profile-schema.json -r my-realm
```

It is documented in the linked API doc but there is no description and I figured `/users/profile` was for updating an individual's profile not the entire declarative user profile schema. Cheers.
